src/dependency-container.ts

```diff
@@ -360,9 +385,14 @@ class InternalDependencyContainer implements DependencyContainer {
     return childContainer;
   }
 
+  public dispose(): void {
```

Comment:
Maybe this could be an async method.
Take for example https://www.npmjs.com/package/redis QUIT command